### PR TITLE
Dynamic Fee Adjustment Crate

### PR DESCRIPTION
**Pull request by gabriele-0201**
This PR extracts the fee adjustment logic into its own crate, requiring multiple modifications because each parameter needs to be passed to the struct instead of being used directly.
 
Open questions:
+ [ ] Is `DynamicFeeAdjustment` an acceptable name?
+ [ ] Is the positioning of the crate within the repository structure acceptable?
+ [ ] Is it ok to define the storage item `NextLengthMultiplier` inside the crate?
 
TODOs:
+ [ ] Fix tests
+ [ ] Restack on top of #157

### CODE REVIEW

**Comment by gabriele-0201** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thrumdev/blobs/161?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #149** <a href="https://app.graphite.dev/github/pr/thrumdev/blobs/149?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #150** <a href="https://app.graphite.dev/github/pr/thrumdev/blobs/150?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #151** <a href="https://app.graphite.dev/github/pr/thrumdev/blobs/151?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #152** <a href="https://app.graphite.dev/github/pr/thrumdev/blobs/152?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #153** <a href="https://app.graphite.dev/github/pr/thrumdev/blobs/153?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #161** <a href="https://app.graphite.dev/github/pr/thrumdev/blobs/161?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/thrumdev/blobs/161?utm_source=stack-comment).

File: sugondat-chain/dynamic-fee-adjustment/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+[package]
+name = "dynamic-fee-adjustment"
```
Comment (rphmeier) [inline sugondat-chain/dynamic-fee-adjustment/Cargo.toml:2]: How about `sugondat-length-fee-adjustment`?

**Comment by rphmeier** [comment]
>  Is the positioning of the crate within the repository structure acceptable?

Seems fine to me! I also considered placing it within the `runtime` directory as it's a runtime utility crate, but we don't have many of those and a flat structure will be easier to navigate until overcrowded.

**Comment by rphmeier** [comment]
Actually, reflecting on the few questions you have brought to mind another approach.

I believe that this should be refactored into a new pallet. There are a few reasons for this:
1. We need a storage item, or perhaps several, for handling parameters. Definitely the `NextLengthMultiplier` would be better as a pallet storage item.
2. We will need some `on_initialize` logic to handle skipped blocks after #157

This also answers the directory location question: it would exist in `pallets`, alongside `blobs`. I recommend for this pallet to require `Config: pallet_transaction_payment::Config`, and it can simply invoke `<Config as pallet_transaction_payment::Config>::MultiplierUpdate` internally to solve the issue of too many generic parameters.

File: sugondat-chain/dynamic-fee-adjustment/Cargo.toml (hunk 1 above)
Comment (gabriele-0201) [reply sugondat-chain/dynamic-fee-adjustment/Cargo.toml]: OK, for now, I used `pallet-sugondat-length-fee-adjustment`. The only thing is that the pallet currently implements both length and weight to the fee. So, `pallet-sugondat-dynamic-fee-adjustment` is probably more general?

**Comment by gabriele-0201** [comment]
The implementation has been moved to #173

**Pull request closed without merging**